packages/dds/migration-shim/src/test/sharedTreeViewTwice.spec.ts

```diff
@@ -0,0 +1,117 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+
+import { type ITestObjectProvider } from "@fluidframework/test-utils";
+import { describeNoCompat } from "@fluid-internal/test-version-utils";
+import {
+	ContainerRuntimeFactoryWithDefaultDataStore,
+	DataObject,
+	DataObjectFactory,
+} from "@fluidframework/aqueduct";
+import { type IChannel } from "@fluidframework/datastore-definitions";
+import {
+	AllowedUpdateType,
+	type ISharedTree,
+	SchemaBuilder,
+	SharedTreeFactory,
+	typeboxValidator,
+	ForestType,
+	type ISharedTreeView2,
+} from "@fluid-experimental/tree2";
+import { type IFluidHandle } from "@fluidframework/core-interfaces";
+
+const treeKey = "treeKey";
+
+class TestDataObject extends DataObject {
+	public async getTree(): Promise<ISharedTree> {
+		const handle: IFluidHandle<IChannel> | undefined =
+			this.root.get<IFluidHandle<IChannel>>(treeKey);
+		assert(handle !== undefined, "No handle found");
+		return (await handle.get()) as ISharedTree;
+	}
+
+	public createTree(type: string): ISharedTree {
+		const channel = this.runtime.createChannel(treeKey, type);
+		this.root.set(treeKey, channel.handle);
+		return channel as ISharedTree;
+	}
+}
+
+const builder = new SchemaBuilder({ scope: "test" });
+const someType = builder.object("abc", {
+	quantity: builder.number,
+});
+
+const schema = builder.intoSchema(someType);
+
+function getNewTreeView(tree: ISharedTree): ISharedTreeView2<typeof schema.rootFieldSchema> {
+	return tree.schematize({
+		initialTree: {
+			quantity: 0,
+		},
+		allowedSchemaModifications: AllowedUpdateType.None,
```

Comment:
@tyler-cai-microsoft - Technically, you're attempting re-schematize a tree that disallows schema updates, but I assume you'll get the same result with 'AllowedUpdateType.Compatible'.

@CraigMacomber - Calling 'schematize(..)' twice back-to-back throws 0x782.

```ts
export function makeTree(context: Context, cursor: ITreeSubscriptionCursor): LazyTreeNode {
	const anchor = cursor.buildAnchor();
	const anchorNode =
		context.forest.anchors.locate(anchor) ??
		fail("cursor should point to a node that is not the root of the AnchorSet");
	const cached = anchorNode.slots.get(lazyTreeSlot);
	if (cached !== undefined) {
		context.forest.anchors.forget(anchor);
		assert(cached.context === context, 0x782 /* contexts must match */);
		return cached;
	}
	const schema = context.schema.nodeSchema.get(cursor.type) ?? fail("missing schema");
	const output = buildSubclass(context, schema, cursor, anchorNode, anchor);
	anchorNode.slots.set(lazyTreeSlot, output);
	anchorNode.on("afterDestroy", cleanupTree);
	return output;
}
```

Reply:
I guess there's a more important question now, if I can't double schematize, how do I get the typed view correctly? If I just start passing around the `ISharedTree` object instead of the `ISharedTreeView2`, what is recommended? https://github.com/microsoft/FluidFramework/pull/18177